### PR TITLE
build: generate declaration files via tsc

### PR DESCRIPTION
**Pull request by nugaon**
It adds `npm run compile:types` command to generate declaration files for each ts file in folder `/src` 

Output dir is `dts`.

### CODE REVIEW

File: tsconfig.d.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "./tsconfig",
+  "compilerOptions": {
+    "strictNullChecks": false,
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    "outDir": "dts",
```
Comment (agazso) [inline tsconfig.d.json:7]: What is the reason to put the .d.ts file in `dts` and not `dist` like the other output files?

Comment (nugaon) [reply tsconfig.d.json:7]: I just separated these two to see what the processes' outputs are exactly, anyway I could put declaration files to the `dist` folder as well.

File: tsconfig.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
   },
   "exclude": [
       "dist",
+      "dts",
```
Comment (agazso) [inline tsconfig.json:16]: We don't need this anymore

**Review by agazso — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 2 above)
Comment (nugaon) [reply tsconfig.json:16]: yeah, it escaped my attention. Thanks!

**Review by agazso — APPROVED** [review]

**Pull request merged**